packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useRef } from 'react';
+
+export type LabelMaskMask = 'phone' | 'ssn' | 'date' | 'day_month';
+
+export interface LabelMaskProps {
+  /**
+   * Must contain a `TextField` component
+   */
+  children: React.ReactNode;
+  labelMask?: LabelMaskMask | ((rawInput: string) => string);
+  // placeholder?: string;
+  // value?: string;
+}
+
+const LabelMask = ({ children, labelMask }: LabelMaskProps) => {
+  const labelMaskPattern = {
+    phone: '[0-9-]*',
+    ssn: '[0-9-*]*',
+    date: '[0-9-]*',
+    day_month: '[0-9.,-]*',
+  };
+
+  /**
+   * Get the child text field. Called as a method so that
+   * updates to the field cause the mask to re-render
+   * @returns {React.ReactElement} Child TextField
+   */
+  const field = (): React.ReactElement => React.Children.only(children as React.ReactElement);
+  const initialValue = field().props.value || field().props.defaultValue || '### ### ###';
+
+  const [value, setValue] = useState(initialValue);
+  const [focus, setFocus] = useState(null);
+
+  /**
+   * @param {Object} evt
+   * @param {React.Element} field - Child TextField
+   */
+  const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+    // Eval use of state here
+    setValue(evt.target.value);
+    // if (typeof field().props.onChange === 'function') {
+    //   value = field().props.onChange(evt);
+    // }
```

Comment:
Having a heck of a time storing the onChange value without saving it to state.

Issue I've encountered so far is not being able to see this text appear over the input as it's typed in.

Reply:
It's up to the parent component to store that value, which means your Storybook story or your doc site example will have to do the state management. This component itself doesn't need any internal state.